openscenario/map_fragment/include/map_fragment/geometry.hpp

```diff
@@ -0,0 +1,306 @@
+    if (radius <= 0.0)
+    {
+      throw std::invalid_argument(
+        "Expected radius to be positive. Actual value:"
+        + std::to_string(radius)
+      );
+    }
+
+    if (!(0 < std::abs(angle) && std::abs(angle) < 2 * M_PI))
+    {
+      throw std::invalid_argument(
+        "Expected angle to be in range (0, 2 * PI) or (-2 * PI, 0). Actual value: "
+        + std::to_string(angle)
+      );
+    }
+
+    center_.y = angle > 0 ? radius : -radius;
+  }
+  
+private:
+  auto getPosition_(double t) -> Point2d override
+  {
+    auto theta = t * angle;
+    Point2d origin = {0, 0};
+    return center_ + rotate(origin - center_, theta);
+  }
+  
+  auto getTangentVector_(double t) -> Vector2d override
+  {
+    auto theta = t * angle;
+    Vector2d v = {1, 0};
+    return rotate(v, theta);
+  }
+};  // class Arc
+
+/**
+ * Combination of two or more component curves, chained after one another
+ */
+class CombinedCurve : public ParametricCurve
+{
+  std::vector<Transformation2d> transformations_;
+
+public:
+  const std::vector<ParametricCurve::Pointer> curves;
+
+  explicit CombinedCurve(std::vector<ParametricCurve::Pointer> const& curves)
+    : curves(curves)
+  {
+    if (curves.size() < 2)
+    {
+      throw std::invalid_argument(
+        "Number of curves to be combined must be two or more. Actual number: "
+        + std::to_string(curves.size())
+      );
+    }
+
+    Transformation2d current_transformation;
+
+    for (const auto& curve : curves)
+    {
+      transformations_.push_back(current_transformation);
+      
+      Point2d origin = {0, 0};
+      Transformation2d local_transformation = {
+        curve->getPosition(1.0) - origin,
+        calculateAngle(curve->getTangentVector(1.0))
+      };
+
+      current_transformation = chainTransformations(current_transformation,
+                                                    local_transformation);
+    }
+  }
+
+private:
+  auto getPosition_(double t) -> Point2d override
+  {
+    auto [curve_id, curve_t] = getCurveIdAndParameter(t);
+    auto curve = curves[curve_id];
+    auto transformation = transformations_[curve_id];
+
+    return applyTransformation(curve->getPosition(curve_t), transformation);
+  }
+
+  auto getTangentVector_(double t) -> Vector2d override
+  {
+    auto [curve_id, curve_t] = getCurveIdAndParameter(t);
+    auto curve = curves[curve_id];
+    auto transformation = transformations_[curve_id];
+
+    return applyTransformation(curve->getTangentVector(curve_t), transformation);
+  }
+
+  auto getCurveIdAndParameter(double t) -> std::pair<std::size_t, double> {
+    validateCurveParameterOrThrow(t);
+    
+    auto range_width_per_curve = 1. / curves.size();
+    auto curve_id = std::min(
+      static_cast<std::size_t>(t / range_width_per_curve),
+      curves.size() - 1);
+    auto curve_t = t / range_width_per_curve - curve_id;
+
+    return std::make_pair(curve_id, curve_t);
+  }
+};  // class CombinedCurve
+
+
+}  // namespace map_fragment
```

Comment:
```suggestion
};  // class CombinedCurve
}  // namespace map_fragment
```

Reply:
Fixed with `ament_clang_format` (although one separating line was left; is that okay?)